src/firebolt/async_db/_types.py

```diff
@@ -362,13 +374,35 @@ def statement_to_set(statement: Statement) -> Optional[SetParameter]:
         and tokens[0].value.lower() == "set"
     ):
         # Check if set statement has a valid format
-        if len(tokens) != 2 or not isinstance(tokens[1], Comparison):
-            raise InterfaceError(
-                f"Invalid set statement format: {statement_to_sql(statement)},"
-                " expected SET <param> = <value>"
+        if len(tokens) == 2 and isinstance(tokens[1], Comparison):
+            return SetParameter(
+                statement_to_sql(tokens[1].left),
+                statement_to_sql(tokens[1].right).strip("'"),
             )
-        return SetParameter(
-            statement_to_sql(tokens[1].left), statement_to_sql(tokens[1].right)
+        # Or if at least there is a comparison
+        cmp_idx = next(
+            (
+                i
+                for i, token in enumerate(tokens)
+                if token.ttype == ComparisonType or isinstance(token, Comparison)
+            ),
+            None,
+        )
+        if cmp_idx:
+            left_tokens, right_tokens = tokens[1:cmp_idx], tokens[cmp_idx + 1 :]
+            if isinstance(tokens[cmp_idx], Comparison):
+                left_tokens = left_tokens + [tokens[cmp_idx].left]
+                right_tokens = [tokens[cmp_idx].right] + right_tokens
```

Comment:
Not sure I understand what this is doing, accounting for `set a>b`?

Reply:
It's mainly accounting for a specific case `set query_parameters={"name":"param1","value":"Hello, world!"}`
This is not considered a correct sql by the parser, so he structure comes a bit weird